### PR TITLE
Problem: use functions and variables from shared libraries

### PR DESCRIPTION
**Pull request by bkrauzova**
Solution: select_asset_element_all_with_warranty_end is used from from fty-common-db
	  FTY_COMMON_LOGGING_DEFAULT_CFG is used from fty-common-logging
	  MLM_ENDPOINT is used from fty-common

Signed-off-by: Barbora Stepankova <BarboraStepankova@eaton.com>

### CODE REVIEW

**Comment by bkrauzova** [comment]
this PR needs new fty-common-db and tests will fail unless PR https://github.com/42ity/fty-common-db/pull/7 is merged and jenkins deployed with image containing the change

File: include/Makefile.am [hunk 1]
```diff
@@ -1,18 +1,16 @@
 ################################################################################
 #  THIS FILE IS 100% GENERATED BY ZPROJECT; DO NOT EDIT EXCEPT EXPERIMENTALLY  #
-#                               NOTE: Modified below!                          #
```
Comment (jimklimov) [inline include/Makefile.am:3]: This. Explicitly. We have a documented procedure to regenerate zproject and reconcile with `git difftool` the inevitable manual changes in some files, marked as such. Follow it.

File: packaging/debian/fty-warranty.install [hunk 2]
```diff
@@ -1,7 +1,5 @@
-### Note: this file was customized for paths
```
Comment (jimklimov) [inline packaging/debian/fty-warranty.install]: And this

File: packaging/redhat/fty-warranty.spec [hunk 3]
```diff
@@ -1,6 +1,5 @@
 #
 #    fty-warranty - Agent sending metrics about warranty expiration
-#    Note: This file was customized after generation; be sure to keep it so
```
Comment (jimklimov) [inline packaging/redhat/fty-warranty.spec]: And this

**Review by jimklimov — CHANGES_REQUESTED** [review]

**Comment by jimklimov** [comment]
Pushed the fixes (reversion of deletion of customized lines, while keeping the valid regenerated changes in those files) to same PR branch, hope that solves the issue. Note the PR should be "squashed" during merging, so git history of those customizations remains in place.

**Review by jana-rapava — APPROVED** [review]

**Comment by jimklimov** [comment]
In the aftermath, thanks all for triggering the issues and investigation into them; a couple of issues in `zproject` and `gsl` are now in code review.

**Pull request merged**